### PR TITLE
Fix assetdetails NP for users using public API

### PR DESCRIPTION
**Pull request by amjad1210**
Fix NP for users using the public API without credentials.

### CODE REVIEW

File: xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java [hunk 1]
```diff
@@ -120,29 +120,35 @@ public void remoteInit() {
...
-          if (!currencies.containsKey(currentCurrencyPair.counter)) {
-            AssetDetail counterAsset =
-                assetDetailMap.get(currentCurrencyPair.counter.getCurrencyCode());
-            BigDecimal counterWithdrawlFee =
-                counterAsset != null ? counterAsset.getWithdrawFee().stripTrailingZeros() : null;
-            currencies.put(
-                currentCurrencyPair.counter,
-                new CurrencyMetaData(counterPrecision, counterWithdrawlFee));
-          }
+
```
Comment (walec51) [inline xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java]: move the declaration in line 83 `Map<String, AssetDetail> assetDetailMap = accountService.getAssetDetails();` to here

**Comment by walec51** [comment]
thanks for fixing this, just one small remark in the review

File: xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java (hunk 1 above)
Comment (amjad1210) [reply xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java]: Hey @walec51 

`Map<String, AssetDetail> assetDetailMap = accountService.getAssetDetails();` is an API call to get all the assets and should only be called once. Placing it on line 123 put it within the for loop which will call it over and over again resulting in a possible ban due to exceeding the rate limit.

Comment (walec51) [reply xchange-binance/src/main/java/org/knowm/xchange/binance/BinanceExchange.java]: ok, your right, didn't notice the loop above

**Pull request merged**